### ISSUE TITLE
Put extensions in subfolder instead main app folder

### ISSUE THREAD

**Issue by rornor** [pyinstaller/pyinstaller#2468]
Hi,

I have `app.spec` file like this:

```python
block_cipher = None

a = Analysis(
    ['app.py'],
    pathex=['C:\\Temp\\test'],
    binaries=[],
    datas=resources,
    hiddenimports=['queue'],
    hookspath=[],
    runtime_hooks=[],
    excludes=[],
    win_no_prefer_redirects=False,
    win_private_assemblies=False,
    cipher=block_cipher,
)

pyz = PYZ(a.pure, a.zipped_data, cipher=block_cipher)

exe = EXE(
    pyz,
    a.scripts,
    exclude_binaries=True,
    name='Test App',
    debug=False,
    strip=False,
    upx=False,
    console=False,
    icon='res/app.ico',
)

coll = COLLECT(
    exe,
    a.binaries,
    a.zipfiles,
    a.datas,
    strip=False,
    upx=False,
    name='Test App',
)
```

as I result I get working executable:

```shell
C:\Temp\test\dist\Test App>dir /b /o
Include
requests
res
tcl
tk
_bz2.pyd
_ctypes.pyd
_hashlib.pyd
_lzma.pyd
_socket.pyd
_ssl.pyd
_tkinter.pyd
base_library.zip
lxml.etree.pyd
MSVCR100.dll
pyexpat.pyd
python34.dll
pywintypes34.dll
select.pyd
Test App.exe
Test App.exe.manifest
tcl86t.dll
tk86t.dll
unicodedata.pyd
win32wnet.pyd
```

I'd prefer to move all extensions (`*.pyd`) in i.e. `ext` subfolder, instead putting all in this base folder.

I couldn't find any information about it in pyinstaller documentation, and was also unable to google for such example. Is this possible and if so please let me know how?

Thanks

**Comment by htgoebel** [comment]
> Is this possible

No, this is not supported.

**Comment by stadelmanma** [comment]
This would probably be satisfied by implementing #1048 as well.

**Comment by rjsdotorg** [comment]
I tried the recipe by @Stevoisiak in #1048 using runtime hooks.
Some .pyd and some dll files could be moved to the /lib folder and be found, but many/most could not.
I do not know the underlying mechanisms well enough to understand why or predict which can be moved to a sub-dir, or how to allow others.
Python27\Scripts\pyinstaller, v3.6

**Comment by Stevoisiak** [comment]
@rjsdotorg You can directly link to the comment with the following url: `https://github.com/pyinstaller/pyinstaller/issues/1048#issuecomment-382050987`. 

As for the hook itself, I got it from [this blogspot post](http://unafaltadecomprension.blogspot.com/2014/07/pyinstaller-separating-executable-from.html). I'm afraid I don't know much more than that.

**Comment by wu933** [comment]
> I tried the recipe by @Stevoisiak in #1048 using runtime hooks.
> Some .pyd and some dll files could be moved to the /lib folder and be found, but many/most could not.
> I do not know the underlying mechanisms well enough to understand why or predict which can be moved to a sub-dir, or how to allow others.
> Python27\Scripts\pyinstaller, v3.6

Have you already solve this problem and give me a suggestion？

**Comment by AutumnSun1996** [comment]
Is there any progress for this?
Tried some methods mentioned here but with no luck.
BTW, after some tries I finally decided to not touch pyinstaller things, and just use a simple wrapper exe outside the dist folder.
Here is my current workaround:
```c
#include<windows.h>
void main(){
    system("lib\\realexe.exe");
}
```

Hope to see a better solution, maybe we can wait for #5841 to be merged and check the result.

**Comment by bwoodsend** [comment]
That wrapper assumes the current working directory. It'll fall to bits if you `cd` into another folder then run it. I'm fairly certain that it will also break if you run the application from explorer rather than `cmd`.

**Comment by AutumnSun1996** [comment]
Yes, the simple code I posted here won't work when called from another folder, but it's not very difficult to add extra support.
Here is a updated one in my project which solved that problem: [wrapper.c](https://github.com/AutumnSun1996/RapidOCR/blob/web-multilanguage/ocrweb_multi/wrapper.c)
But after all, it's a very simple workaround, and likely not suitable for more complex needs.
